src/commonMain/kotlin/exchange.dydx.abacus/state/manager/notification/NotificationsProvider.kt

```diff
@@ -0,0 +1,69 @@
+package exchange.dydx.abacus.state.manager.notification
+
+import exchange.dydx.abacus.output.Notification
+import exchange.dydx.abacus.protocols.ParserProtocol
+import exchange.dydx.abacus.state.manager.V4Environment
+import exchange.dydx.abacus.state.manager.notification.providers.BlockRewardNotificationProvider
+import exchange.dydx.abacus.state.manager.notification.providers.OrderStatusChangesNotificationProvider
+import exchange.dydx.abacus.state.model.TradingStateMachine
+import exchange.dydx.abacus.utils.IMap
+import exchange.dydx.abacus.utils.JsonEncoder
+import exchange.dydx.abacus.utils.ParsingHelper
+import exchange.dydx.abacus.utils.UIImplementations
+import kollections.toIMap
+
+interface NotificationsProviderProtocol {
+    fun buildNotifications(
+        subaccountNumber: Int
+    ): IMap<String, Notification>
+}
+
+class NotificationsProvider(
+    private val stateMachine: TradingStateMachine,
+    private val uiImplementations: UIImplementations,
+    private val environment: V4Environment,
+    private val parser: ParserProtocol,
+    private val jsonEncoder: JsonEncoder,
+    private val useParentSubaccount: Boolean = false,
+    private val providers: List<NotificationsProviderProtocol> = listOf(
```

Comment:
I would remove this default value, so that we can remove the transitive dependencies from `NotificationsProvider` (`stateMachine`, `uiImplementations`, `environment`, `jsonEncoder`, `parser`, etc)